sdk/iOS/test/MSSyncTableTests.m

```diff
@@ -340,6 +403,146 @@ -(void) testInsertPushInsertPush
     
 }
 
+
+
+-(void) testInsertUpdateWhilePushing
```

Comment:
Ideally we'd end up with 2 unit tests: update with server changes (leaves version alone), this test case (Server item == sent copy), and a new one for pending delete gets version updated (if server == sent).